src/libraries/Microsoft.PowerFx.Interpreter/EvalVisitor.cs

```diff
@@ -244,6 +244,34 @@ public override FormulaValue Visit(BinaryOpNode node, SymbolContext context)
                     return OperatorGtTime(this, context, node.IRContext, args);
                 case BinaryOpKind.GeqTime:
                     return OperatorGeqTime(this, context, node.IRContext, args);
+                case BinaryOpKind.DynamicGetField:
+                    if (arg1 is CustomObjectValue cov && arg1 is StringValue sv)
+                    {
+                        if (cov.Impl.IsObject)
+                        {
+                            if (cov.Impl.TryGetProperty(sv.Value, out var res))
+                            {
+                                return new CustomObjectValue(node.IRContext, res);
+                            }
+                            else
+                            {
+                                return new BlankValue(node.IRContext);
+                            }
+                        }
+                        else
+                        {
+                            return new ErrorValue(node.IRContext, new ExpressionError()
+                            {
+                                Message = "Accessing a field is not valid on this value",
+                                Span = node.IRContext.SourceContext,
+                                Kind = ErrorKind.BadLanguageCode
+                            });
+                        }
+                    }
+                    else
+                    {
+                        return new BlankValue(node.IRContext);
```

Comment:
> return new BlankValue(node.IRContext); [](http://example.com/codeflow?start=23&length=39)

Can this happen? Or would this be a compiler error caught at an earlier binding stage (ie, should we throw here?)

Reply:
This should probably be split into cases. if Arg1 is blankvalue, we should absolutely return blankvalue, but other than that, we should probably thow InvalidOperation

Reply:
same for passing through error values from arg1, I think